openpype/hosts/traypublisher/plugins/publish/collect_csv_ingest_instance_data.py

```diff
@@ -0,0 +1,36 @@
+from pprint import pformat
+import pyblish.api
+from openpype.pipeline import publish
+
+
+class CollectCSVIngestInstancesData(
+    pyblish.api.InstancePlugin,
+    publish.OpenPypePyblishPluginMixin,
+    publish.ColormanagedPyblishPluginMixin
+):
+    """Collect CSV Ingest data from instance.
+    """
+
+    label = "Collect CSV Ingest instances data"
+    order = pyblish.api.CollectorOrder + 0.1
+    hosts = ["traypublisher"]
+    families = ["csv"]
+
+    def process(self, instance):
+        self.log.info(f"Collecting {instance.name}")
+
+        # expecting [(colorspace, repre_data), ...]
+        prepared_repres_data_items = instance.data[
+            "prepared_data_for_repres"]
+
+        for colorspace, repre_data in prepared_repres_data_items:
+            # only apply colorspace to those which are not marked as thumbnail
+            if colorspace != "_thumbnail_":
```

Comment:
I understand the potential need for this - but seems like an odd hardcoded value without any documentation elsewhere.

Also, aren't there other ways to mark a representations as a thumbnail. E.g. wasn't there some sort of representations tags for that or whatever?

Reply:
yes, this is only to differentiate between representation data captured from CSV rows which are not thumbnail related. Since the colorspace column is defined explicitly a colorspace, but there is nothing on thumbnail raw, the decision was made to add `hardcoded keyword` so it is during this particular loop avoided for colorspace data set. See https://github.com/ynput/OpenPype/pull/5795/files/c33a666037035523be60a2dff8c09f5f1fc50a11#diff-6dfa88e411c00bbd9ce1d8681bbe38dcdabc86c48e55f7137cd024a962f602b3R291